test/framework/robot.py

```diff
@@ -567,6 +567,14 @@ def test_get_toolchain_hierarchy(self):
             {'name': 'gompi', 'version': '1.4.10'},
```

Comment:
gompi is a bit of a broken choice when you try to add dummy, because the underlying GCC doesn't actually have any dependencies on anything built with `dummy` which is (currently) required. I guess we should hard-include `dummy` when the relevant option is there